src/dev_server/index.tsx

```diff
@@ -1,46 +1,53 @@
-import React, {useEffect} from 'react';
+import React, {useEffect, useRef, useState} from 'react';
 import {Platform} from 'react-native';
 import {Config, SocketMessage} from '../types';
 import {ScriptManager, Federated} from '@callstack/repack/client';
 import NetInfo from '@react-native-community/netinfo';
-import dgram from 'react-native-udp';
+import TcpSocket from 'react-native-tcp-socket';
 import axios from 'axios';
 
 let config: Config;
+const RETRY_TIMER = 5000;
 
 const DevServer = props => {
-  const onSocket = (handler) => msg => {
-    const json = JSON.parse(msg.toString());
-    if (json?._connected) {
-      return;
-    }
-
-    // We should have a context object now
-    const context = new Proxy(json, handler);
+  const connection = useRef<TcpSocket.Socket>();
+  const partialMessage = useRef('');
+  const _retryTimer = useRef<NodeJS.Timeout>();
 
-    props.onContextChanged(context);
-  };
-
-  const onError = err => {
-    console.error('[Sleeper] Socket error: ' + err);
+  const _onConnected = async (value: boolean) => {
+    props.onConnected(value);
   };
 
-  const bindSocket = async socket => {
-    const netInfo = await NetInfo.fetch();
-    const netInfoDetails = netInfo?.details;
+  const onSocket = (handler) => msg => {
+    let json;
+    try {
```

Comment:
what can happen if we don't try?

Reply:
This is used to reconstruct partial messages sent over several socket events - failing the JSON.parse() is how we know an incomplete packet was sent.